src/lib/y2network/interface.rb

```diff
@@ -46,5 +72,23 @@ def ==(other)
     # eql? (hash key equality) should alias ==, see also
     # https://ruby-doc.org/core-2.3.3/Object.html#method-i-eql-3F
     alias_method :eql?, :==
+
+    def type
+      Yast::NetworkInterfaces.GetType(name)
+    end
+
+    def config
+      system_config(name)
+    end
+
+  private
+
+    def system_config(name)
+      Yast::NetworkInterfaces.devmap(name)
```

Comment:
so basically hash, so we do not plan yet to integrate new config object, right?

Reply:
basically what we use now. If I start playing with config objects then we get super big PR2 which is not what I want ;-)